### PR TITLE
Bug/MDS-1353 Remove double toast messages on Minespace login

### PR DESCRIPTION
**Pull request by james-hollinger**
In the authorization wrapper for Minspace, the component checks to see if you have a JWT token and makes a getUserInfo call if you do. However if the token is invalid (such as if it's expired if you return to the site after 24hrs), an error is returned [toast message #1], and the catch handler for getUserInfo logs the user out [toast message #2].

This PR changes the auth flow slightly to handle this case, and expands error handling slightly for other failure cases.

 - getUserInfo now accepts an error message, or re-throws errors if none are supplied. 
- unAuthenticateUser accepts a toast message, and only shows the toast if supplied
- authenticateUser now unAuthenticates if it catches an error.

### CODE REVIEW

File: frontend-public/src/tests/actionCreators/authenticationActionCreator.spec.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ describe("`getUserInfoFromToken` action creator", () => {
 
   it("Request failure, dispatches `error` with correct response", () => {
     mockAxios.onGet(url, MOCK.createMockHeader()).reply(400, MOCK.ERROR);
-    return getUserInfoFromToken()(dispatch).then(() => {
+    return getUserInfoFromToken("A Token", "An Error Message")(dispatch).then(() => {
```
Comment (nayr974) [inline frontend-public/src/tests/actionCreators/authenticationActionCreator.spec.js]: "A Verbose String"

File: frontend-public/src/actionCreators/authenticationActionCreator.js [hunk 2]
```diff
@@ -44,19 +50,20 @@ export const authenticateUser = (code) => (dispatch) => {
     redirect_uri: ENV.BCEID_LOGIN_REDIRECT_URI,
     client_id: ENV.KEYCLOAK.clientId,
   };
-  dispatch(request(reducerTypes.AUTHENTICATE_USERR));
+  dispatch(request(reducerTypes.AUTHENTICATE_USER));
```
Comment (nayr974) [inline frontend-public/src/actionCreators/authenticationActionCreator.js]: THIS_IS_GOODERR

File: frontend-public/src/HOC/AuthenticationGuard.js [hunk 3]
```diff
@@ -15,7 +15,9 @@ export const AuthenticationGuard = (isPublic) => (WrappedComponent) => {
     componentDidMount() {
       const token = localStorage.getItem("jwt");
       if (token && !this.props.isAuthenticated) {
-        this.props.getUserInfoFromToken(token);
+        this.props.getUserInfoFromToken(token).catch(() => {
+          // Silently fail
```
Comment (nayr974) [inline frontend-public/src/HOC/AuthenticationGuard.js]: I like to create a const doNothing = () => {}; as it makes itself describing without comments.

File: frontend-public/src/actionCreators/authenticationActionCreator.js (hunk 2 above)
Comment (james-hollinger) [reply frontend-public/src/actionCreators/authenticationActionCreator.js]: US  IS  YOU
ER

File: frontend-public/src/HOC/AuthenticationGuard.js (hunk 3 above)
Comment (james-hollinger) [reply frontend-public/src/HOC/AuthenticationGuard.js]: Fair, but seems a waste if its only used once. Maybe we should add it to constants.js?

Comment (NWCalvank) [reply frontend-public/src/HOC/AuthenticationGuard.js]: Lodash provides a `noop` function 🤷‍♂️

Also, we use `() => {}` all the time as a default in `defaultProps`, so we'll definitely be able to use this constant or lodash function many times

**Pull request merged**